src/Config/Validator/Deploy/ReportDirNestingLevel.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Config\Validator\Deploy;
+
+use Magento\MagentoCloud\Config\Environment;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+use Magento\MagentoCloud\Filesystem\ConfigFileList;
+use Magento\MagentoCloud\Config\Validator\ResultFactory;
+use Magento\MagentoCloud\Filesystem\Driver\File;
+use Symfony\Component\Serializer\Encoder\XmlEncoder;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Filesystem\FileSystemException;
+use Symfony\Component\Serializer\Exception\NotEncodableValueException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Validates the value specified for the directory nesting level conifgured for error reporting
+ */
+class ReportDirNestingLevel implements ValidatorInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var Environment
+     */
+    private $environment;
+
+    /**
+     * @var ConfigFileList
+     */
+    private $configFileList;
+
+    /**
+     * @var File
+     */
+    private $file;
+
+    /**
+     * @var XmlEncoder
+     */
+    private $encoder;
+
+    /**
+     * @var ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param Environment $environment
+     * @param ConfigFileList $configFileList
+     * @param File $file
+     * @param XmlEncoder $encoder
+     * @param ResultFactory $resultFactory
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        Environment $environment,
+        ConfigFileList $configFileList,
+        File $file,
+        XmlEncoder $encoder,
+        ResultFactory $resultFactory
+    ) {
+        $this->logger = $logger;
+        $this->environment = $environment;
+        $this->configFileList = $configFileList;
+        $this->file = $file;
+        $this->encoder = $encoder;
+        $this->resultFactory = $resultFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        $envVarValue = $this->environment->getEnvVarMageErrorReportDirNestingLevel();
+        $reportConfigFile = $this->configFileList->getErrorReportConfig();
+
+        try {
+            if (false !== $envVarValue) {
```

Comment:
so, you decided to leave validation on the deploy phase which is not related to cloud?

Reply:
@NadiyaS I removed the code that was responsible for validation, leaving only a check for the existence of the environment variable `MAGE_ERROR_REPORT_DIR_NESTING_LEVEL`. This is necessary to evaluate the case when the report directory nesting level not setting is not used.